src/main/scala/Decoder.scala

```diff
@@ -26,116 +25,112 @@ class Decoder(bitWidth: Int = 32) extends Module {
         // Loads
-        BitPat("b?????????????????000?????0000011")  -> List(INST_I,      LB, false.B,   false.B, true.B,    false.B,  true.B,  false.B),
-        BitPat("b?????????????????001?????0000011")  -> List(INST_I,      LH, false.B,   false.B, true.B,    false.B,  true.B,  false.B),
-        BitPat("b?????????????????100?????0000011")  -> List(INST_I,     LBU, false.B,   false.B, true.B,    false.B,  true.B,  false.B),
-        BitPat("b?????????????????101?????0000011")  -> List(INST_I,     LHU, false.B,   false.B, true.B,    false.B,  true.B,  false.B),
-        BitPat("b?????????????????010?????0000011")  -> List(INST_I,      LW, false.B,   false.B, true.B,    false.B,  true.B,  false.B),
+        BitPat("b?????????????????000?????0000011")  -> BitPat(Cat(INST_I.asUInt,       LB.asUInt, false.B,   false.B, true.B,    false.B,  true.B,  false.B)),
+        BitPat("b?????????????????001?????0000011")  -> BitPat(Cat(INST_I.asUInt,       LH.asUInt, false.B,   false.B, true.B,    false.B,  true.B,  false.B)),
+        BitPat("b?????????????????100?????0000011")  -> BitPat(Cat(INST_I.asUInt,      LBU.asUInt, false.B,   false.B, true.B,    false.B,  true.B,  false.B)),
+        BitPat("b?????????????????101?????0000011")  -> BitPat(Cat(INST_I.asUInt,      LHU.asUInt, false.B,   false.B, true.B,    false.B,  true.B,  false.B)),
+        BitPat("b?????????????????010?????0000011")  -> BitPat(Cat(INST_I.asUInt,       LW.asUInt, false.B,   false.B, true.B,    false.B,  true.B,  false.B)),
         // Stores
-        BitPat("b?????????????????000?????0100011")  -> List(INST_S,      SB, false.B,   false.B, true.B,    false.B, false.B,   true.B),
-        BitPat("b?????????????????001?????0100011")  -> List(INST_S,      SH, false.B,   false.B, true.B,    false.B, false.B,   true.B),
-        BitPat("b?????????????????010?????0100011")  -> List(INST_S,      SW, false.B,   false.B, true.B,    false.B, false.B,   true.B),
+        BitPat("b?????????????????000?????0100011")  -> BitPat(Cat(INST_S.asUInt,       SB.asUInt, false.B,   false.B, true.B,    false.B, false.B,   true.B)),
+        BitPat("b?????????????????001?????0100011")  -> BitPat(Cat(INST_S.asUInt,       SH.asUInt, false.B,   false.B, true.B,    false.B, false.B,   true.B)),
+        BitPat("b?????????????????010?????0100011")  -> BitPat(Cat(INST_S.asUInt,       SW.asUInt, false.B,   false.B, true.B,    false.B, false.B,   true.B))
       )
-    ) // format: on
+    ,
+                                                        BitPat(Cat(IN_ERR.asUInt, ERR_INST.asUInt, false.B,  false.B, false.B,    false.B, false.B,  false.B)) // Default values
+      // format: on
+      ),
+    )
+    .asTypeOf(new DecType)
 
-  io.DecoderPort.rd       := 0.U
-  io.DecoderPort.rs1      := 0.U
-  io.DecoderPort.rs2      := 0.U
-  io.DecoderPort.imm      := 0.S
```

Comment:
Defaulting to 0 introduce an extra Mux which is useless.

Reply:
Nice, had to adjust the expected bogus values in my tests for this change.